### PR TITLE
Update Kerberos docs and make some api internal

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fixes #

**Changes proposed in this request**

**Testing**

**Performance impact**

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/Kerberos/TicketCacheReader.cs [hunk 1]
```diff
@@ -8,12 +8,10 @@
 
 namespace Microsoft.Identity.Client.Platforms.Features.DesktopOs.Kerberos
 {
-
-#pragma warning disable 618 // This workaround required for Native Win32 API call
     /// <summary>
     /// Helper class to check Kerberos Ticket in user's Ticket Cache.
     /// </summary>
-    public class TicketCacheReader : IDisposable
+    internal class TicketCacheReader : IDisposable
```
Comment (neha-bhargava) [inline src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/Kerberos/TicketCacheReader.cs:14]: Isn't this a breaking change?

Comment (pmaytak) [reply src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/Kerberos/TicketCacheReader.cs:14]: Looks like these were supposed to be internal and no one's supposed to be using it. KerberosSupplementalTicketManager has the public logic.

File: src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs [hunk 2]
```diff
@@ -340,8 +340,12 @@ public PublicClientApplicationBuilder WithParentActivityOrWindow(Func<IntPtr> wi
         /// Sets the parameters required to get a Kerberos Ticket from Azure AD service.
         /// </summary>
         /// <param name="servicePrincipalName">Service principal name to get Kerberos Service Ticket.</param>
-        /// <param name="ticketContainer">Container to use for Kerberos Ticket.</param>
-        /// <returns>The builder to chain the .With methods</returns>
+        /// <param name="ticketContainer">Specify where the Kerberos ticket will be returned - as a claim in the id token or as a claim in the access token. 
+        /// If the ticket is for the client application, use the IdToken. If the ticket is for the downstream api, use the access token.</param>
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs:344]:
```suggestion
        /// <param name="ticketContainer">Specify where the Kerberos ticket will be returned - as a claim in the ID token or as a claim in the access token. 
        /// If the ticket is for the client application, use the ID token. If the ticket is for the downstream API, use the access token.</param>
```

File: src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs [hunk 3]
```diff
@@ -340,8 +340,12 @@ public PublicClientApplicationBuilder WithParentActivityOrWindow(Func<IntPtr> wi
...
         /// </summary>
         /// <param name="servicePrincipalName">Service principal name to get Kerberos Service Ticket.</param>
-        /// <param name="ticketContainer">Container to use for Kerberos Ticket.</param>
-        /// <returns>The builder to chain the .With methods</returns>
+        /// <param name="ticketContainer">Specify where the Kerberos ticket will be returned - as a claim in the id token or as a claim in the access token. 
+        /// If the ticket is for the client application, use the IdToken. If the ticket is for the downstream api, use the access token.</param>
+        /// <remaraks>
+        /// The expiry of the Kerberos ticket is tied to the expiry of the token that contains it.
+        /// MSAL provides several helper APIs to read and write Kerberos tickets from the Windows Ticket Cache <see cref="KerberosSupplementalTicketManager"/>
+        /// </remaraks>
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs:348]:
```suggestion
        /// <remarks>
        /// The expiry of the Kerberos ticket is tied to the expiry of the token that contains it.
        /// MSAL provides several helper APIs to read and write Kerberos tickets from the Windows Ticket Cache - see <see cref="KerberosSupplementalTicketManager"/>.
        /// </remarks>
```

**Review by pmaytak — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/Kerberos/TicketCacheReader.cs (hunk 1 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/Kerberos/TicketCacheReader.cs:14]: +1 @neha-bhargava and @pmaytak - I forgot to update this.

**Review by neha-bhargava — APPROVED** [review]

**Review by trwalke — APPROVED** [review]

**Pull request merged**